package/MDAnalysis/core/selection.py

```diff
@@ -228,24 +228,17 @@ def apply(self, group):
 class DistanceSelection(Selection):
     """Base class for distance search based selections
 
-    Grabs the flags for this selection
-     - 'use_KDTree_routines'
-     - 'use_periodic_selections'
-
     Populates the `apply` method with either
      - _apply_KDTree
      - _apply_distmat
     """
-    def __init__(self):
-        if flags['use_KDTree_routines'] in (True, 'fast', 'always'):
+    def __init__(self, periodic, kdtree):
+        if kdtree:
             self.apply = self._apply_KDTree
         else:
             self.apply = self._apply_distmat
 
-        self.periodic = flags['use_periodic_selections']
-        # KDTree doesn't support periodic
-        if self.periodic:
```

Comment:
So interestingly because periodic was the default, this line meant that KDTree was never used